### PR TITLE
Update FindTarantool.cmake

### PR DESCRIPTION
**Pull request by filonenko-mikhail**
Update FindTarantool to use with static bundled tarantool

There is canonical https://github.com/tarantool/modulekit/blob/ckit/FindTarantool.cmake

### CODE REVIEW

**Comment by Totktonada** [comment]
Hi! Please, describe your problem briefly.

File: cmake/FindTarantool.cmake [hunk 1]
```diff
@@ -9,7 +9,8 @@ macro(extract_definition name output input)
 endmacro()
 
 find_path(TARANTOOL_INCLUDE_DIR tarantool/module.h
-  HINTS ENV TARANTOOL_DIR /usr/local/include
```
Comment (tsafin) [inline cmake/FindTarantool.cmake:12]: Why have you removed /usr/local/include from consideration? What is the order supposed to be used if there is _no_ `$TARANTOOL_DIR` defined in the environment?

Comment (filonenko-mikhail) [reply cmake/FindTarantool.cmake:12]: I think cmake internal hard-coded find_path hints

**Comment by Totktonada** [comment]
It seems, it follows the fix of https://github.com/tarantool/modulekit/issues/2 ([patch](https://github.com/tarantool/modulekit/commit/676344bd0eb60cf708b36eab9e78b01af37a22bd)), but misses the rockspec update.

**Comment by filonenko-mikhail** [comment]
No way here

**Pull request closed without merging**